common/djangoapps/student/tests/test_models.py

```diff
@@ -200,6 +205,34 @@ def test_upgrade_deadline_with_schedule_and_professional_mode(self):
         self.assertIsNotNone(enrollment.schedule)
         self.assertIsNone(enrollment.upgrade_deadline)
 
+class PendingNameChangeTests(SharedModuleStoreTestCase):
+    """
+    Tests the deletion of PendingNameChange records
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(PendingNameChangeTests, cls).setUpClass()
+        cls.user = UserFactory()
+        cls.user2 = UserFactory()
+
+    def setUp(self):
+        self.name_change, _ = PendingNameChange.objects.get_or_create(
+            user=self.user,
+            new_name='New Name PII',
+            rationale='for testing!'
+        )
+
+    def test_delete_by_user_removes_pending_name_change(self):
+        self.assertEqual(1, len(PendingNameChange.objects.all()))
```

Comment:
Can you move `self.assertEqual(1, len(PendingNameChange.objects.all()))` into the setUp() method? Then you can prevent the first line of each test from repeating.

Reply:
Good suggestion. Will do that